.github/workflows/container-image-build.yml

```diff
@@ -0,0 +1,70 @@
+name: build-images action template
+permissions: {}
+
+on:
+  workflow_call:
+    inputs:
+      image-name:
+        required: true
+        description: "Name of the image to build"
+        type: string
+      dockerfile-directory:
+        required: false
+        description: "The directory where the dockerfile locates, as relative to the repo root"
+        type: string
+        default: .
+      pushImage:
+        required: false
+        description: "Whether to push the image afterwards"
+        type: boolean
+        default: true
+
+jobs:
+  job:
+    runs-on: ubuntu-latest
+    if: github.repository.fork == false
```

Comment:
Just a thought: if this is not specifically checking for metal3-io/project-infra, can any non-fork repository call this workflow and push stuff with it? Why are checking for forks only?

/hold

Reply:
It shouldn't check for `metal3-io/project-infra`, because the `github.repository` will be the caller's repository (`metal3-io/baremetal-operator`, for e.g.)

Technically any repo can call this template, but the secrets from this repo won't be exposed to anyone. Afaiu, workflow template works just like a go/python package, any script can import it, but the caller needs to provide the necessary environment for it to run.

Reply:
OK, thanks for clarifying it. Makes sense, but you cannot always be sure with Github :)

Reply:
/unhold

Reply:
@tuminoid I changed the logic now, so that it will only be triggered by repos inside `metal3-io`. Please notice as well that `project-infra` repo doesn't have access to the newly-added secrets `QUAY_USERNAME` and `QUAY_PASSWORD`.